lib/cylc/config.py

```diff
@@ -1193,31 +1196,32 @@ def configure_queues(self):
                             queues[self.Q_DEFAULT]['members'].remove(qmember)
                         except ValueError:
                             if qmember in requeued:
-                                msg = "%s: ignoring '%s' (%s)" % (
-                                    key, qmember, 'task already assigned')
+                                msg = "%s: ignoring %s (%s)" % (
+                                    key, qmember,
+                                    'already assigned to a queue')
                                 warnings.append(msg)
                             elif qmember not in all_task_names:
-                                msg = "%s: ignoring '%s' (%s)" % (
+                                msg = "%s: ignoring %s (%s)" % (
                                     key, qmember, 'task not defined')
                                 warnings.append(msg)
                             else:
                                 # Ignore: task not used in the graph.
                                 pass
                         else:
+                            myq[key] = qmember
                             qmembers.append(qmember)
                             requeued.append(qmember)
-
-            if warnings:
-                err_msg = "Queue configuration warnings:"
-                for msg in warnings:
-                    err_msg += "\n+ %s" % msg
-                LOG.warning(err_msg)
-
             if qmembers:
                 queue['members'] = qmembers
             else:
                 del queues[key]
 
+        if warnings:
+            err_msg = "Queue configuration warnings:"
+            for msg in warnings:
+                err_msg += "\n+ %s" % msg
+            LOG.warning(err_msg)
+
```

Comment:
(This block was wrongly inside the outermost loop, resulting in repeated warnings!)